src/commands/elf-symbols/__tests__/elf.test.ts

```diff
@@ -0,0 +1,763 @@
+          sh_link: 0,
+          sh_name: 17,
+          sh_offset: BigInt(6610),
+          sh_size: BigInt(281),
+          sh_type: SectionHeaderType.SHT_STRTAB,
+        },
+      ])
+    })
+  })
+
+  describe('readElfProgranHeaderTable', () => {
+    test('return program header table for ELF file', async () => {
+      const reader = await createReaderFromFile(`${fixtureDir}/dyn_aarch64`)
+      const elfResult = await readElfHeader(reader)
+      const programHeaders = await readElfProgramHeaderTable(reader, elfResult.elfHeader!)
+
+      expect(programHeaders).toEqual([
+        {
+          p_align: BigInt(65536),
+          p_filesz: BigInt(1592),
+          p_flags: 5,
+          p_memsz: BigInt(1592),
+          p_offset: BigInt(0),
+          p_paddr: BigInt(0),
+          p_type: ProgramHeaderType.PT_LOAD,
+          p_vaddr: BigInt(0),
+        },
+        {
+          p_align: BigInt(65536),
+          p_filesz: BigInt(488),
+          p_flags: 6,
+          p_memsz: BigInt(496),
+          p_offset: BigInt(3632),
+          p_paddr: BigInt(69168),
+          p_type: ProgramHeaderType.PT_LOAD,
+          p_vaddr: BigInt(69168),
+        },
+        {
+          p_align: BigInt(8),
+          p_filesz: BigInt(384),
+          p_flags: 6,
+          p_memsz: BigInt(384),
+          p_offset: BigInt(3648),
+          p_paddr: BigInt(69184),
+          p_type: ProgramHeaderType.PT_DYNAMIC,
+          p_vaddr: BigInt(69184),
+        },
+        {
+          p_align: BigInt(4),
+          p_filesz: BigInt(36),
+          p_flags: 4,
+          p_memsz: BigInt(36),
+          p_offset: BigInt(456),
+          p_paddr: BigInt(456),
+          p_type: ProgramHeaderType.PT_NOTE,
+          p_vaddr: BigInt(456),
+        },
+        {
+          p_align: BigInt(4),
+          p_filesz: BigInt(52),
+          p_flags: 4,
+          p_memsz: BigInt(52),
+          p_offset: BigInt(1392),
+          p_paddr: BigInt(1392),
+          p_type: ProgramHeaderType.PT_GNU_EH_FRAME,
+          p_vaddr: BigInt(1392),
+        },
+        {
+          p_align: BigInt(16),
+          p_filesz: BigInt(0),
+          p_flags: 6,
+          p_memsz: BigInt(0),
+          p_offset: BigInt(0),
+          p_paddr: BigInt(0),
+          p_type: ProgramHeaderType.PT_GNU_STACK,
+          p_vaddr: BigInt(0),
+        },
+        {
+          p_align: BigInt(1),
+          p_filesz: BigInt(464),
+          p_flags: 4,
+          p_memsz: BigInt(464),
+          p_offset: BigInt(3632),
+          p_paddr: BigInt(69168),
+          p_type: ProgramHeaderType.PT_GNU_RELRO,
+          p_vaddr: BigInt(69168),
+        },
+      ])
+    })
+  })
+
+  describe('isSupportedArch', () => {
+    test('return true for supported arch', () => {
+      expect(isSupportedArch('aarch64')).toBeTruthy()
+      expect(isSupportedArch('x86_64')).toBeTruthy()
+    })
+    test('return false for unsupported arch', () => {
+      expect(isSupportedArch('arm')).toBeFalsy()
```

Comment:
For the profiling use case we currently support only arm64 and amd64.
`objcopy` does not support easily architecture other than the host one, I have to use a generic architecture to make it accept the input file and then overwrite the header, and I was not sure it would work well with a different endianness.
I added support for ARM as well as some big endian / little endian test cases.